### PR TITLE
added docker compose for production and development setups

### PR DESCRIPTION
**Pull request by sreevardhanreddi**
This PR adds

1. docker-compose setup for local development and production deployment
2. made seeding data to database independent of the api server, to seed data to database run ```python bhagavad_gita_api/cli.py seed-data```
3. added CD setup for deployment when the code is merged to the **main** branch

### CODE REVIEW

**Comment by samanyougarg** [comment]
@sreevardhanreddi Do you mind running the pre-commit hooks before pushing? The checks seem to be failing because of that. Here are the instructions for the same: https://github.com/gita/bhagavad-gita-api/blob/main/.github/CONTRIBUTING.md

**Comment by sreevardhanreddi** [comment]
> @sreevardhanreddi Do you mind running the pre-commit hooks before pushing? The checks seem to be failing because of that. Here are the instructions for the same: https://github.com/gita/bhagavad-gita-api/blob/main/.github/CONTRIBUTING.md

Done 😃

**Comment by samanyougarg** [comment]
Do you mind adding some instructions to `readme.md` and `contributing.md` as well? New users may not be able to understand how to use it.

**Comment by samanyougarg** [comment]
@sreevardhanreddi I have tried running the dev one using the command:
```shell
 docker-compose -f docker-compose.dev.yml up
```
but the API doesn't seem to be starting up:
Screenshot 2021-07-08 at 8 31 27 PM

Am I doing something wrong here?

File: bhagavad_gita_api/config.py [hunk 1]
```diff
@@ -4,6 +4,19 @@
 from pydantic import AnyUrl, BaseSettings
```
Comment (samanyougarg) [inline bhagavad_gita_api/config.py:9]: Can you please import and load dotenv? 
Currently, it doesn't seem to be able to load the values from `.env`.

```python
from dotenv import load_dotenv

load_dotenv()
```

**Review by samanyougarg — COMMENTED** [review]
Works well locally 👌

**Comment by samanyougarg** [comment]
We have a GitHub workflow (https://github.com/gita/bhagavad-gita-api/blob/main/.github/workflows/publish.yml) that builds and pushes the image to DockerHub on every release? 
Can we make use of that so that whenever there's a new release, we build and push the image to DockerHub and then pull and run it on our DO instance? Would love to hear your thoughts.

**Review by samanyougarg — APPROVED** [review]

**Review by sanujsood — APPROVED** [review]

**Pull request merged**